### PR TITLE
Relax deps

### PR DESCRIPTION
**Pull request by gadomski**
## What I am changing

- Relax the dependencies and switch to **geozero** v0.14 from a fork
- Update the docs link to the released spec

This shades on personal preference, but as a library-first crate, I prefer to relax the version requirements and trust SemVer a bit more. `Cargo.lock` still lets CI and others build exactly like we build, but it lets downstreams have a bit more flexibility.

It also reduces the maintenance burden of keeping deps up to date.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -4,19 +4,19 @@ version = "0.1.0"
...
-pest = "2.7.11"
-pest_derive = "2.7.11"
-serde = "1.0.204"
-serde_derive = "1.0.204"
-thiserror = "1.0.63"
-serde_json = { version = "1.0.120", features = ["preserve_order"] }
+geozero = "0.14"
+boon = "0.6"
+lazy_static = "1"
+pest = "2"
```
Comment (kylebarron) [inline Cargo.toml:10]: I think changing `2.7.x` to `2` is especially risky, because in theory there are _new features_ in the minor releases from `2` to `2.7` (and not just bug fixes from `2.7` to `2.7.11`). And so there's no way to tell in isolation whether we use any of the features added in `2.2` or `2.5` that didn't exist in `2.0.0`.

**Review by kylebarron — APPROVED** [review]

**Comment by gadomski** [comment]
> The advice I've heard before is to set the versions in Cargo.toml to the earliest version with which you've tested. 

I _think_ that makes sense except that "I made code in my library that depends on a PATCH release" feels unlikely, I guess? A PATCH should (usually) be transparent to a downstream, which is why "1.2" feels more like the actual requirement to me. But I don't really have a strong preference — I just find that many dependency checking tools throw a lot of noise when we go all the way to PATCH. But maybe that's a tool problem to solve, not something to tweak in our dependences?

Basically, don't have a strong opinion, other than that I agree that "1" is probably bad 👍🏼.

**Comment by kylebarron** [comment]
> > The advice I've heard before is to set the versions in Cargo.toml to the earliest version with which you've tested.
> 
> I _think_ that makes sense except that "I made code in my library that depends on a PATCH release" feels unlikely, I guess?

That doesn't seem that unlikely if you rely on a bug fix in a patch release? It doesn't _always_ matter to pin to a min patch version in semver, but sometimes it really does matter, and you can't tell which case is which just by looking at a `Cargo.toml` in isolation.
